agent/src/softwarecontaineragent.cpp

```diff
@@ -172,22 +172,24 @@ SoftwareContainerAgent::SoftwareContainerPtr SoftwareContainerAgent::makeSoftwar
     return container;
 }
 
-ContainerID SoftwareContainerAgent::createContainer(const std::string &config)
+bool SoftwareContainerAgent::createContainer(const std::string &config, ContainerID& containerID)
 {
     profilepoint("createContainerStart");
     profilefunction("createContainerFunction");
 
     if (!parseConfig(config)) {
         log_error() << "The provided config could not be successfully parsed";
-        return -1;
+        containerID = -1;
+        return false;
     }
 
     std::pair<ContainerID, SoftwareContainerPtr> pair = std::move(getContainerPair());
     pair.second->setMainLoopContext(m_mainLoopContext);
 
     if (isError(pair.second->init())) {
         log_error() << "Could not init the container";
-        return -1;
+        containerID = -1;
```

Comment:
Actually there is already a defined INVALID_CONTAINER_ID in softwarecontainer-common.h